docs/hooks/intro.md

```diff
@@ -0,0 +1,132 @@
+# Hooks
+
+`react-plot` provides hook that allows to add functionalities to the plot.
+
+To benefit of those features the hooks have to be wrapped in a `PlotController`.
+
+```jsx live
+function Demo() {
+  return (
+    <PlotController>
+      <Zoomable />
+    </PlotController>
+  );
+
+  function Zoomable() {
+    const zoom = useRectangularZoom();
+    return (
+      <Plot width={300} height={300}>
+        <LineSeries
+          data={[
+            { x: 1, y: 1 },
+            { x: 2, y: 2 },
+            { x: 3, y: 3 },
+            { x: 4, y: 2 },
+            { x: 5, y: 1 },
+          ]}
+        />
+        <Annotations>{zoom.annotations}</Annotations>
+      </Plot>
+    );
+  }
+}
+```
+
+```jsx live
+function Demo() {
+  const [currentEvent, setCurrentEvent] = useState(undefined);
+
+  return (
+    <PlotController>
+      <Tracking />
+      <EventInfo />
+    </PlotController>
+  );
+
+  function EventInfo() {
+    return (
+      <>
+        {currentEvent && (
```

Comment:
You should generally not declare components inside components (here `EventInfo` and `Tracking` are nested in `Demo`). This results in the components being recreated every time Demo is rendered (when the state changes). When a new component replaces another one, React will completely unmount the DOM, which triggers cleanup effects, resets the internal states, and may also impact the events emitted by the browser

Reply:
Thanks. I found finally the correct way (I think) to do it with docusaurus and react-live.